File: InterviewPratice/Javascript/firstNotRepeatingCharacter.js
const firstNotRepeatingCharacter = (s) => {
    let a = [];
    
    for(let i = 0; i < 26; i++) a.push(0);
    
    for(let e of s) a[e.charCodeAt(0) - 97]++;

    for(let e of s) if(a[e.charCodeAt(0) - 97] == 1) return e;
    
    return "_";
}

File: InterviewPratice/Java/FirstNotRepeatingCharacter.java
final class FirstNotRepeatingCharacter {

	char firstNotRepeatingCharacter(String s) {
	    int[] a = new int[26];
	    
	    for(char c : s.toCharArray()) a[c - 'a']++;
	    
	    for(char c : s.toCharArray()) if(a[c - 'a'] == 1) return c;
	    
	    return '_';
	}
}